troubleshoot-stale-read.md

```diff
@@ -0,0 +1,215 @@
+    stopped: false,
+```
+
+The preceding output helps you determine:
+
+- Whether locks are blocking resolved-ts.
+- Whether the apply index is too small to update safe-ts.
+- Whether the leader is sending a sufficiently updated resolved-ts when a follower peer exists.
+
+## Troubleshooting tips
+
+### Handle slow transaction commit
+
+A transaction that takes a long time to commit is often a large transaction. The prewrite phase of this slow transaction leaves some locks, but it takes too long before the commit phase clean the locks. To troubleshoot, you can try to identify the transaction to which the locks belong and try to know the reason they exist, such as using logs.
+
+The following list some actions you can take:
+
+- Specify `--log` option in the `tikv-ctl` command and check TiKV logs to find the specific locks with their start_ts.
+- Search the start_ts in both TiDB and TiKV logs to identify issues with the transaction.
+
+    If a query takes over 60 seconds, an `expensive_query` log is printed with the SQL statement. You can use the start_ts value to match the log. The following is an example:
+
+    ```log
+    [2023/07/17 19:32:09.403 +08:00] [WARN] [expensivequery.go:145] [expensive_query] [cost_time=60.025022732s] [cop_time=0.00346666s] [process_time=8.358409508s] [wait_time=0.013582596s] [request_count=278] [total_keys=9943616] [process_keys=9943360] [num_cop_tasks=278] [process_avg_time=0.030066221s] [process_p90_time=0.045296042s] [process_max_time=0.052828934s] [process_max_addr=192.168.31.244:20160] [wait_avg_time=0.000048858s] [wait_p90_time=0.00006057s] [wait_max_time=0.00040991s] [wait_max_addr=192.168.31.244:20160] [stats=t:442916666913587201] [conn=2826881778407440457] [user=root] [database=test] [table_ids="[100]"] [**txn_start_ts**=442916790435840001] [mem_max="2514229289 Bytes (2.34 GB)"] [sql="update t set b = b + 1"]
+    ```
+
+- Use the [`CLUSTER_TIDB_TRX`](/information-schema/information-schema-tidb-trx.md#cluster_tidb_trx) table to find active transactions if you cannot get enough information about the locks from logs.
+- Execute [`SHOW PROCESSLIST`](/sql-statements/sql-statement-show-processlist.md) to view the current sessions connected to the same TiDB server and their time spent on current statement. But it does not show start_ts.
+
+If the locks exist due to ongoing large transactions, consider modifying your application logic as these locks can hinder the progress of resolve-ts.
+
+If the locks do not belong to any ongoing transactions, it might be due to a coordinator (TiDB) crashing after it prewrites the locks. In this case, TiDB will automatically resolve the locks. No action is required unless the problem persists.
+
+### Handle long-lived transactions
+
+Transactions that remain active for a long time could possibly block the advance of resolved-ts, even if they eventually commit quickly. This is because it's the start-ts of these long-lived transactions that are used to calculate the resolved-ts.
```

Comment:
```suggestion
Transactions that remain active for a long time could possibly block the advance of resolved-ts, even if they eventually commit quickly. This is because it is the start-ts of these long-lived transactions that is used to calculate the resolved-ts.
```